meshnets/modules/processor.py

```diff
@@ -1,10 +1,40 @@
-""""Define the GraphProcessor class."""
+""""Define the Processor and ProcesserLayer classes."""
 
+import torch
+from torch.nn import Identity
+from torch_geometric.data import Batch
+from torch_geometric.nn.conv import MessagePassing
 
-class GraphProcessor:
 
-    def __init__(self, *args):
-        pass
+class MGNProcessorLayer(MessagePassing):
+    """Single MGN Message Passing layer for a Graph object."""
 
-    def __call__(self, *args):
-        pass
+    def __init__(self, latent_size):
+        super().__init__()
+
+        # TODO(victor) : implement layer
+        self.layer = Identity(latent_size)
+
+    def forward(self, graph: Batch) -> Batch:
+        return self.layer(graph)
+
+
+class MGNProcessor(torch.nn.Module):
+    """MGN Processor for graphs.
+    
+    It is a sequence of message-passing MGNProcessorLayers applied to the
+    the latent node and mesh features."""
+
+    def __init__(self, latent_size, message_passing_steps):
+        """Initialize the processor given the latent size
+        and number of message passing steps"""
+        super().__init__()
+
+        # TODO(victor) : implement processor
+        self.processor = Identity(latent_size, message_passing_steps)
+
+    def forward(self, graph: Batch) -> Batch:
+        """Apply the GraphProcessor to a batch of graphs.
+        
+        Return the batch with processed features."""
+        return self.processor(graph)
```

Comment:
The prediciton shape in our case is 1 feature per node. In the input, graph.x has 3 features, so I reduce that with a linear layer in the dummy decoder: (nb_nodes, 3) -> (nb_nodes, 1). We could indeed simplify the model even more by including the linear layer in the main module, or even with `return graph.x[:, 0]`.

However, I think that using the composed model structure we already have offers a good idea of what the model is going to look like at the end. This way, I was also able to check that the input/output types of each module works well with our dataset. At the end, the total model only has 4 parameters in the decoder, so it is very lightweight but should still allow us to observe some behavior during training.

Reply:
Btw, MGNProcessor is only the Processor part of our MeshGraphNet model, and for the moment it takes in a graph batch and return a graph batch, but this might change as we go.